src/store/reducers/registerReducer.ts

```diff
@@ -0,0 +1,40 @@
+export type UserType = {
+  email: string;
+  password: string;
+};
+
+type InitialStateType = {
+  isAuth: boolean;
+  user: UserType;
+};
+
+const initialState = {
+  isAuth: false,
+  user: {} as UserType,
```

Comment:
try to create a initial values for field like `user: { name: '', email: '' }` etc.